src/python.rs

```diff
@@ -117,16 +127,6 @@ impl<'j> PythonParser<'j> {
                 }
                 Ok(dict.to_object(py))
             }
-            _ => {
```

Comment:
Well, removing this clause did cause errors to fire here about non-exhaustive matching :wink:. But I take your point about it not enforcing all datatypes.

I think part of the performance win in #48 was that it effectively removed a large set of branches (from `Peak::new`) because we have these same set of branches anyway when we decide what to do with the peek's `u8`. Put another way, we assume that the peek is always valid and defer erroring until after we fail to parse an integer.

It's definitely possible that there's another way to express this which keeps the same performance property, might require a play and a think.